src/middlewared/middlewared/plugins/system_advanced/config.py

```diff
@@ -259,6 +261,13 @@ async def do_update(self, data):
             if original_data['motd'] != config_data['motd']:
                 await self.middleware.call('etc.generate', 'motd')
 
+            if original_data['login_banner'] != config_data['login_banner']:
+                with open('/etc/login_banner', 'w', encoding='utf-8') as f:
```

Comment:
This is an async function so we should avoid file handling in these

Reply:
I think you can move this to etc plugin and specify another mako file where you do this and add the perms there too - it will handle applying them nicely as well

Reply:
What if we put the file generation in our initial `sshd_config.mako` instead? I feel like having a whole new .mako just to write one string is a little much. We already do some file IO in this config too

Reply:
That should work